src/Splitio/Services/Common/SyncManager.cs

```diff
@@ -94,8 +98,31 @@ public void Shutdown()
                 Task.WaitAll(task.ToArray(), Constants.Gral.DestroyTimeount);
 
                 _synchronizer.ClearFetchersCache();
+            }
+            catch (Exception ex)
+            {
+                _log.Error($"Somenthing went wrong destroying the SDK.", ex);
+            }
+        }
+
+        public async Task ShutdownAsync()
+        {
+            try
+            {
+                _telemetryRuntimeProducer.RecordSessionLength(CurrentTimeHelper.CurrentTimeMillis() - _startSessionMs);
+
+                _ctsStreaming.Cancel();
+                _ctsStreaming.Dispose();
 
-                _log.Info("SDK has been destroyed.");
+                await _synchronizer.StopPeriodicDataRecordingAsync();
+                await _synchronizer.StopPeriodicFetchingAsync();
+                await _onStreamingStatusTask.StopAsync();
+                await _sseHandler.StopWorkersAsync();
+                await _pushManager.StopAsync();
+                await _startupTask.StopAsync();
+                await _tasksManager.DestroyAsync();
```

Comment:
Would it make sense to run these concurrently? Or at least some, there may be dependencies between the tasks that I'm not aware of.
```suggestion
                List<Task> synchronizerTasks = new List<Task>
                {
                    _synchronizer.StopPeriodicDataRecordingAsync(),
                    _synchronizer.StopPeriodicFetchingAsync(),
                    _onStreamingStatusTask.StopAsync(),
                    _sseHandler.StopWorkersAsync(),
                    _pushManager.StopAsync(),
                    _startupTask.StopAsync()
                };
                
                await Task.WhenAll(destroyTasks);
                await _tasksManager.DestroyAsync()
```